### DISCUSSION TITLE
Trying to lists applets but get "could not SELECT default selected: 0x6700"

### DISCUSSION THREAD

**Discussion by drov0** [martinpaljak/GlobalPlatformPro#331] [Q&A]
Hello, 

I managed through other means to install a few applets, now I'd like to list all the applets on my card, and thought I'd do that with gpPro:

but I get:
```
java -jar gp.jar -l
Could not SELECT default selected: 0x6700 (Wrong length (Lc))
```

in fact I get that with most of the commands:

```
java -jar gp.jar -acr-list
Could not SELECT default selected: 0x6700 (Wrong length (Lc))
```

Anyone encountered that before or could point me in a direction to fix it ?

**Comment by martinpaljak** [comment]
Please provide:
- full log (-dvi added)
- type of card etc

**Comment by drov0** [comment]
Hello ! 

```
GlobalPlatformPro v20.01.23-0-g5ad373b
Running on Linux 6.2.0-26-generic amd64, Java 11.0.20 by Ubuntu
# Detected readers from JNA2PCSC
[*] Identiv SCR3500 A Contact Reader [CCID Interface] (54302118606805) 00 00
[ ] Alcor Micro AU9540 01 00
SCardConnect("Identiv SCR3500 A Contact Reader [CCID Interface] (54302118606805) 00 00", T=*) -> T=0, 3B9F96801F878031E073FE211B674A4C753030056E88
SCardBeginTransaction("Identiv SCR3500 A Contact Reader [CCID Interface] (54302118606805) 00 00")
Reader: Identiv SCR3500 A Contact Reader [CCID Interface] (54302118606805) 00 00
ATR: 3B9F96801F878031E073FE211B674A4C753030056E88
More information about your card:
    http://smartcard-atr.appspot.com/parse?ATR=3B9F96801F878031E073FE211B674A4C753030056E88

A>> T=0 (4+0000) 00A40400 00 
A<< (0000+2) (58ms) 6700
Could not SELECT default selected: 0x6700 (Wrong length (Lc))
```

As for the type of card I'm not 100% sure, it's an engineering sim card from workz (https://www.workz.com/)

**Comment by martinpaljak** [comment]
Try explicitly selecting AID "A0 00 00 01 51 00 00 00" (-sdaid). Or maybe it has a global platform ISD under some other AID. Check the card documentation. Generally speaking, SIM cards and GlobalPlatform don't always play along (even if they are javacard capable). If you provide cards and docs I might work on them.

How did you install those applets? Whatever the tooling, I guess it also allows to list any applets.

**Comment by drov0** [comment]
```
java -jar gp.jar -sdaid A000000151000000
Could not SELECT Security Domain: 0x6A82 (Application/file not found)
```
Huh, seems like it's under some other AID, I'll try to check where it is. 

I basically built from scratch my own gpPro alternative in python because I never got the authentication keys directly, so I'm sending apdus by simulating OTA calls (and also because I'm a beginner in the world of smart card programming so it felt better to build from scratch to learn how things work under the hood).

Thanks for your input ! I think I will try to look at your code and see how I can replicate it :)

**Comment by martinpaljak** [comment]
See https://github.com/martinpaljak/GlobalPlatformPro/issues/91.

I don't work with SIM cards (operators usually provide ways to reach their SIM cards, so little benefit for homelab or dev environment)

**Comment by drov0** [comment]
Yeah I think that's totally fair ! I don't mind reimplementing, I was just hoping I could fast track things.

Thanks for your time :) your resources are great (I compile my javacard code using https://github.com/martinpaljak/ant-javacard).

**Comment by martinpaljak** [comment]
If you have time to describe OTA framing and necessary parameters so that I would not have to read the 3gpp docs, please contribute, might add it to gppro as well.
